cmudb/exploration/util.py

```diff
@@ -0,0 +1,95 @@
+import signal
+import subprocess
+import sys
+from enum import Enum
+from typing import List, Tuple, AnyStr
+from typing import Union
+
+# Project (relative to exploration directory)
+PROJECT_ROOT = "../.."
+ENV_FOLDER = "../env"
+
+# Docker
+CONTAINER_BIN_DIR = "/home/terrier/repo/build/bin"
+DOCKER_VOLUME_DIR = "/mnt/docker/volumes"
+IMAGE_TAG = "pgnp"
+EXPLORATION_CONTAINER_NAME = "exploration"
+EXPLORATORY_COMPOSE = "docker-compose-exploration.yml"
+EXPLORATORY_PROJECT_NAME = "exploratory"
+
+# Postgres
+REPLICA_PORT = 15722
+EXPLORATION_PORT = 42666
+
+# ZFS
+ZFS_DOCKER_VOLUME_POOL = "zpool-docker/volumes"
+REPLICA_VOLUME_POOL = "pgdata-replica"
+EXPLORATION_VOLUME_POOL = "pgdata-exploration"
+ZFS_SNAPSHOT_NAME = "explore"
+
+# Misc
+UTF_8 = "utf-8"
+
+
+class OutputStrategy(Enum):
+    Capture = (subprocess.PIPE, subprocess.PIPE)
+    Print = (sys.stdout, sys.stderr)
+    Hide = (subprocess.DEVNULL, subprocess.DEVNULL)
+
+
+def execute_sys_command(cmd: Union[str, List[str]],
```

Comment:
I have very mixed feelings about wrapping `subprocess.Popen`. On the one hand I find `subprocess.Popen` a bit cumbersome to use by itself. On the other hand it seems like all of our projects wrap the method in slightly different ways, and they often get a bit messy.